Marlin/ultralcd.cpp

```diff
@@ -575,29 +575,31 @@ static void lcd_status_screen() {
 
   #if ENABLED(BABYSTEPPING)
 
-    int babysteps_done = 0;
+    long babysteps_done = 0;
+    float distance_done = 0.0f;
 
     static void _lcd_babystep(const AxisEnum axis, const char* msg) {
       ENCODER_DIRECTION_NORMAL();
       if (encoderPosition) {
-        int distance = (int32_t)encoderPosition * BABYSTEP_MULTIPLICATOR;
+        int babystep_increment = (int32_t)encoderPosition * BABYSTEP_MULTIPLICATOR;
         encoderPosition = 0;
         lcdDrawUpdate = LCDVIEW_REDRAW_NOW;
-        thermalManager.babystep_axis(axis, distance);
-        babysteps_done += distance;
+        thermalManager.babystep_axis(axis, babystep_increment);
+        babysteps_done += babystep_increment;
+        distance_done = (float) ( (1000 * babysteps_done) / planner.axis_steps_per_mm[axis] ) / 1000.0f;// 1000 to print 3 Dec Places
       }
-      if (lcdDrawUpdate) lcd_implementation_drawedit(msg, itostr3sign(babysteps_done));
+      if (lcdDrawUpdate) lcd_implementation_drawedit(msg, ftostr43sign(distance_done));
```

Comment:
If you do this here instead…
```cpp
if (lcdDrawUpdate) {
  float distance_done = 0.001f * ((1000 * babysteps_done) / planner.axis_steps_per_mm[axis]);
  lcd_implementation_drawedit(msg, ftostr43sign(distance_done));
}
```
…then you don't need to define `distance_done` above, or set it to `0.0` below.